pydrive2/auth.py

```diff
@@ -273,12 +273,13 @@ def ServiceAuth(self):
     user_email = self.client_config.get('client_user_email')
     service_email = self.client_config['client_service_email']
     file_path = self.client_config['client_pkcs12_file_path']
-    self.credentials = ServiceAccountCredentials.from_p12_keyfile(
-        service_account_email=service_email,
-        filename=file_path,
-        scopes=scopes)
-    if user_email:
-        self.credentials = self.credentials.create_delegated(sub=user_email)
+    if file_path:
```

Comment:
could you please elaborate on this? why do we need this check?

Reply:
if `file_path` is not specified it shouldn't try to load p12 key from unspecified path and error out. The idea is that encrypted key and secret to access service account already might be stored in presaved credentials file generated from previous app run.

Reply:
so, it's still required for the first run?

Reply:
in this case this method does not do anything at all?

Reply:
These changes were made to allow to run gdrive tests on Travis only with specified GDRIVE_USER_CREDENTIALS_DATA secured env var and service account email in settings.
 
> in this case this method does not do anything at all?

You are right. Except validation of available settings by call of `self.LoadServiceConfigSettings()` it seems doesn't do anything else. 
I guess `if file_path:` can be moved right after call of `self.LoadServiceConfigSettings()`

Reply:
> so, it's still required for the first run?

Yes, it is required for the first run (except DVC specific scenario if GDRIVE_USER_CREDENTIALS_DATA env var is set from the first run ). Might makes sense to check there if credentials file can be found at all to not proceed further if it doesn't exist yet.

Reply:
if it does not do anything, why is it being called at all?

Reply:
@shcheklein I've added `return` in case if credentials were already loaded from the file. Should work now in similar way as auth for usual Google account.